### ISSUE TITLE
Inconsistent `calculated_noise_multiplier` when using `gradient_accumulation_steps` in Gemma3 LoRA example

### ISSUE THREAD

**Issue by SunnierLee** [google-deepmind/jax_privacy#234]
Hi JAX Privacy team,

Thank you for maintaining this excellent library. I am currently exploring the `examples/dp_sgd_keras_gemma3_lora_finetuning_samsum.ipynb` example and have encountered a discrepancy regarding how `train_steps` and `gradient_accumulation_steps` affect the noise multiplier calculation.

### Context
In the notebook, it is stated:
> "Also note that `gradient_accumulation_steps` is provided as a separate parameter. JAX Privacy takes it into account to calculate the real number of optimization steps during the training."

Based on this, I understood that if I provide the total number of *micro-batch* steps as `train_steps` (i.e., `EPOCHS * (TRAIN_SIZE // BATCH_SIZE)`), the library should internally divide this by `gradient_accumulation_steps` to determine the actual number of *optimizer updates* (macro-steps) for the privacy accounting.

### Observed Behavior
I ran two experiments with the same effective batch size ($1024$) and privacy budget ($\epsilon=4.0, \delta=2e-05$), but different physical batch sizes and gradient accumulation steps. The dataset size is $14,732$.

**Configuration 1:**
*   `epoch`: 3
*   `batch_size`: 4
*   `gradient_accumulation_steps`: 256
*   `effective_batch_size`: $4 \times 256 = 1024$
*   `train_steps` (passed): $3 \times (14732 // 4) = 11,049$ (Total micro-steps)
*   **Output:**
    ```text
    Calculating noise multiplier for: self.epsilon=4.0, self.delta=2e-05, self.effective_batch_size=1024, self.train_steps=11049, self.train_size=14732.
    Finished calculating noise multiplier: calculated_noise_multiplier=7.6917...
    ```

**Configuration 2:**
*   `epoch`: 3
*   `batch_size`: 16
*   `gradient_accumulation_steps`: 64
*   `effective_batch_size`: $16 \times 64 = 1024$
*   `train_steps` (passed): $3 \times (14732 // 16) = 2,760$ (Total micro-steps)
*   **Output:**
    ```text
    Calculating noise multiplier for: self.epsilon=4.0, self.delta=2e-05, self.effective_batch_size=1024, self.train_steps=2760, self.train_size=14732.
    Finished calculating noise multiplier: calculated_noise_multiplier=3.9159...
    ```

### The Issue
If `JAX Privacy` correctly accounts for `gradient_accumulation_steps` to derive the real number of optimization steps, both configurations should result in the **same** noise multiplier. It appears that the privacy accountant might be using the raw `train_steps` value directly without dividing by `gradient_accumulation_steps`, or there is a misunderstanding in how `train_steps` should be passed when gradient accumulation is enabled.

Any guidance or clarification would be greatly appreciated. Thanks!

**Comment by Anu27n** [comment]
I'd like to work on this. The accountant is already correct (`train_steps` is the number of optimizer updates, as documented in #267); the Gemma examples were still passing micro-batch counts, which is what produced the inconsistent noise multipliers.

Fix is in #372: the notebooks and Keras API docs now set `train_steps = epochs * (train_size // effective_batch_size)`, plus a calibration test that the noise multiplier is invariant to the physical-batch / accumulation split.
